core/error_macros.h

```diff
@@ -115,27 +115,40 @@ extern bool _err_error_exists;
 #endif
 
 #define ERR_FAIL_INDEX(m_index, m_size)                                                                                    \
-	do {                                                                                                                   \
+	{                                                                                                                      \
```

Comment:
what we could do is to restore the do {} while() function, but add a comment of what it is for, so this does not happen again